exchange/websocket/manager.go

```diff
@@ -682,11 +681,23 @@ func (m *Manager) shutdown() error {
 	// flush any subscriptions from last connection if needed
 	m.subscriptions.Clear()
 
-	m.setState(disconnectedState)
-
 	close(m.ShutdownC)
+	m.setState(disconnectedState)
 	m.Wg.Wait()
 	m.ShutdownC = make(chan struct{})
+
+	// Field connections are persistent once set, so we need to reassign the shutdown channel
+	for _, conn := range []Connection{m.Conn, m.AuthConn} {
+		if conn == nil {
+			continue
+		}
+		underlying, ok := conn.(*connection)
```

Comment:
🚧 Change request:
This isn't actually the underlying though.
Naming of the range var is a bit up in the air.
I rarely want to do this to an interface type like this, so I'd fall back to treating it like an `any` when naming. `connInterface` or `connIface` works too
```suggestion
		conn, ok := connA.(*connection)
```